test/runtime/runtime_test.go

```diff
@@ -25,37 +25,66 @@ func (s *testingRuntimeSuite) TestPrependEnvsPrefix(c *C) {
 		comment     string
 		cmd         string
 		env         map[string]string
+		soft        bool
 		expectedCmd string
 		err         string
 	}{
 		{
 			"no variable in environment",
 			"/node server.js",
 			map[string]string{},
+			false,
 			"/node server.js",
 			"",
 		},
 		{
 			"single variable in environment",
 			"/node server.js",
 			map[string]string{"PORT": "8000"},
+			false,
 			"--env=PORT=8000 /node server.js",
 			"",
 		},
 		{
 			"two variables in environment",
 			"/node server.js",
 			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
+			false,
 			// Order is not guaranteed hence the following regex is needed:
 			"(--env=PORT=8000 --env=ENDPOINT=foo.com|--env=ENDPOINT=foo.com --env=PORT=8000) /node server.js",
 			"",
 		},
+		{
+			"no variable in environment - soft",
+			"/node server.js",
+			map[string]string{},
+			true,
+			"/node server.js",
+			"",
+		},
+		{
+			"single variable in environment - soft",
+			"/node server.js",
+			map[string]string{"PORT": "8000"},
+			true,
+			"--env=PORT?=8000 /node server.js",
+			"",
+		},
+		{
+			"two variables in environment - soft",
+			"/node server.js",
+			map[string]string{"PORT": "8000", "ENDPOINT": "foo.com"},
+			true,
+			// Order is not guaranteed hence the following regex is needed:
+			"(--env=PORT?=8000 --env=ENDPOINT=foo.com|--env=ENDPOINT=foo.com --env=PORT=8000) /node server.js",
```

Comment:
Is there a particular reason for expecting `?` only in one case? Shouldn't all use `?=` since the test is using `soft` setting?